tools/harness-sniffer/OT_Sniffer.py

```diff
@@ -0,0 +1,135 @@
+'''OpenThread Sniffer API implementation'''
+
+import subprocess
+from GRLLibs.UtilityModules.ModuleHelper import ModuleHelper
+from ISniffer import ISniffer
+
+
+class OT_Sniffer(ISniffer):
+    common_speed = [460800, 115200, 9600]
+
+    def __init__(self, **kwargs):
+        try:
+            self.channel = kwargs.get('channel')
+            self.port = kwargs.get('addressofDevice')
+            self.baudrate = None
+            self.subprocess = None
+            self.is_active = False
+
+        except Exception, e:
+            ModuleHelper.WriteIntoDebugLogger('OT_Sniffer: [intialize] --> ' + str(e))
+
+    def discoverSniffer(self):
+        sniffers = []
+
+        p_discover = subprocess.Popen('extcap_ot.bat --extcap-interfaces', stdout=subprocess.PIPE, shell=True)
+        for line in p_discover.stdout.readlines():
+           if line.startswith('interface'):
+               try:
+                   interface_port = line[line.index('value=')+6: line.index('}{display')]
+                   sniffers.append(OT_Sniffer(addressofDevice=interface_port, channel=ModuleHelper.Default_Channel))
+               except Exception, e:
+                   ModuleHelper.WriteIntoDebugLogger('OT_Sniffer: [discoverSniffer] --> Error: ' + str(e))
+
+        p_discover.wait()
+        return sniffers
+
+    def startSniffer(self, channelToCapture, captureFileLocation):
+        """
+        Method for starting the sniffer capture on given channel and this should create wireshark 'pcapng' file at the given location. Capture should happen in background so that method call will be non-blocking and asynchronous.
+        @param channelToCapture : int : channel number to start the capture
+        @param captureFileLocation : string : Full path with the filename with extension is passed.
+        """
+        try:
+            # baudrate auto detection
+            cmd_baudrate = ['extcap_ot.bat', '--extcap-interface', self.port,'--extcap-baudrate']
+            p_baudrate = subprocess.Popen(cmd_baudrate, stdout=subprocess.PIPE, shell=True)
+
+            for line in p_baudrate.stdout.readlines():
+                if (line.startswith('baudrate')):
+                    self.baudrate = line.split(':')[1][:6]
+
+            p_baudrate.wait()
+
+            # start sniffer
+            if self.baudrate is not None:
+                self.setChannel(channelToCapture)
+                p_where = subprocess.Popen('where python', stdout=subprocess.PIPE, shell=True)
+                pythonExe = p_where.stdout.readline().replace("\r\n", "")
+
+                if pythonExe.endswith(".exe"):
+                    snifferPy = pythonExe[:-10]+'Scripts\sniffer.py'
```

Comment:
add comment with example of what the pythonExe looks like? not easy to read hard-coded number

Reply:
should use `os.path.xxxxx` methods to manipulate path